core/src/wsv.rs

```diff
@@ -157,6 +159,68 @@ impl WorldStateView {
         tokens
     }
 
+    /// Returns a set of permission tokens granted to this account separately.
+    pub fn account_own_permission_tokens(
+        &self,
+        account: &Account,
+    ) -> impl ExactSizeIterator<Item = PermissionToken> {
+        self.world
+            .account_permission_tokens
+            .get(account.id())
+            .map_or_else(Default::default, |permissions_ref| {
+                permissions_ref.value().clone()
+            })
+            .into_iter()
+    }
+
+    /// Return `true` if `Account` contains permission token not accociated with any role
+    #[inline]
+    pub fn account_contains_own_permission(
+        &self,
+        account: &<Account as Identifiable>::Id,
+        token: &PermissionToken,
+    ) -> bool {
+        self.world
+            .account_permission_tokens
+            .get_mut(account)
+            .map_or(false, |permissions| permissions.contains(token))
+    }
+
+    /// Add [`permission`](PermissionToken) into the [`Account`].
+    ///
+    /// If `Account` did not have this permission present, `true` is returned.
+    /// If `Account` did have this permission present, `false` is returned.
+    pub fn add_account_permission(
```

Comment:
I don't check if account with such id exist, but this seems unnecessary, because we use `add/remove_account_permission` in such a way that this check happened before (except tests).

And such check will introduce additional branching and more complicated api.

Reply:
Such checks should be done outside of this method, just like `wsv.modify_account()` does at the moment